joomla/joomla-cms
# Fix Showon feature for fields in a subform field, AND speadup showon code

### Summary of Changes

The pull fix Showon feature for a fields in the subform field, #11548 . 
Also here is the perfomance optimisation.

Perfomance of showon feature in global configuration:
**before** 134ms
![screen 2016-10-22 17 17 50 574x306](https://cloud.githubusercontent.com/assets/1568198/19620016/8780a60e-987c-11e6-94b7-de0691ced67f.png)

**after** 59ms
![screen 2016-10-22 17 18 20 668x288](https://cloud.githubusercontent.com/assets/1568198/19620021/947ff45e-987c-11e6-9292-afa92ce78048.png)
### Testing Instructions

Create subform field (example in module Custom):

``` xml
<field name="items" type="subform" formsource="modules/mod_custom/test.xml"
  multiple="true"  label="Subform Field" />
```

Create a form (`modules/mod_custom/test.xml`) for the subform field, that contain `showon` feature:

``` xml
<?xml version="1.0" encoding="UTF-8"?>
<form>
<field name="type" type="list" default="0" label="Type" >
        <option value="0">Small</option>
        <option value="1">Medium</option>
        <option value="2">Large</option>
    </field>

        <field name="something"  type="text"
            label="Show when Type=Large" showon="type:2" />
</form>
```

or even use XML from #8524 (that even better for test, but without the radio button with "btn-group-yesno" as it still broken).

And make sure that `showone` feature works.
### Documentation Changes Required

none
